splink/cost_of_blocking_rules.py

```diff
@@ -0,0 +1,112 @@
+import logging
+from typing import Dict, List
+
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+
+def calculate_field_freedom_cost(combination_of_brs: List[Dict]) -> int:
+    """
+    We want a higher scores (lower cost) for combinations of blocking rules that allow
+    as much variation in each field as possible
+
+    e.g.  we don't like combinations of four rules
+    that hold first_name and surname constant in 3 out of 4
+    and only allows them to vary in one, even if that affords greater
+    variance to other fields.
+
+    That is, we would prefer a spread in which fields are held fixed across the blocking
+    rules
+
+    Calculates the field cost for a given combination of brs. It counts the number of
+    times each field is allowed to vary (i.e., not included in the blocking rules).
+
+    Args:
+        combination_of_brs (List[Dict]): The combination_of_brs rows
+
+    Returns:
+        int: The field freedom cost.
+    """
+
+    total_cost = 0
+    field_names = [c for c in combination_of_brs[0].keys() if c.startswith("__fixed__")]
+
+    for field in field_names:
+        field_can_vary_count = sum(row[field] == 0 for row in combination_of_brs)
+
+        costs_by_count = {0: 20, 1: 10, 2: 2, 3: 1, 4: 1}
```

Comment:
If this is hard coded, it should live above the loop, so we don't continuously reconstruct it.

Reply:
It would be useful to know where this has come from too (apologies if it's stated somewhere, I'm polishing this up before eating).

Reply:
Done